core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -657,10 +657,10 @@ class GroupMetadataManager(brokerId: Int,
       // special handling for version 0:
       // set the expiration time stamp as commit time stamp + server default retention time
       val updatedOffsetAndMetadata =
-        if (offsetAndMetadata.expireTimestamp == org.apache.kafka.common.requests.OffsetCommitRequest.DEFAULT_TIMESTAMP)
-        offsetAndMetadata.copy(expireTimestamp = offsetAndMetadata.commitTimestamp + config.offsetsRetentionMs)
-      else
-        offsetAndMetadata
+        if (offsetAndMetadata.expireTimestamp.contains(org.apache.kafka.common.requests.OffsetCommitRequest.DEFAULT_TIMESTAMP))
+          offsetAndMetadata.copy(expireTimestamp = Some(offsetAndMetadata.commitTimestamp + config.offsetsRetentionMs))
```

Comment:
Is there a strong reason to use copy rather than just constructing a new object?

Reply:
This is not my addition. Isn't this more concise than the alternative?
```
          new OffsetAndMetadata(offsetAndMetadata.offsetMetadata,
            offsetAndMetadata.commitTimestamp, 
            expireTimestamp = Some(offsetAndMetadata.commitTimestamp + config.offsetsRetentionMs))
```
I can construct a new object if you think that's better practice.